Source/Scene/LabelCollection.js

```diff
@@ -79,27 +81,91 @@ define([
+    var glyphBuffer = 2;    // Whitespace buffer around a glyph in pixels
+    var glyphRadius = 8;    // How many pixels around the glyph shape to use for encoding distance
+    var glyphCutoff = 0.25;  // How much of the radius (relative) is used for the inside part the glyph
+
+    function getCSSValue(element, property) {
+        return document.defaultView.getComputedStyle(element,null).getPropertyValue(property);
+    }
+
+    var fontInfoCache = {};
+    /**
+     * Given a CSS font string return an object containing the fontFamily, fontSize, fontStyle and fontWeight.
+     */
+    function getFontInfo(font) {
+        var fontInfo = fontInfoCache[font];
+        if (fontInfo) {
+            return fontInfo;
         }
 
-        writeTextToCanvasParameters.fill = style === LabelStyle.FILL || style === LabelStyle.FILL_AND_OUTLINE;
-        writeTextToCanvasParameters.stroke = style === LabelStyle.OUTLINE || style === LabelStyle.FILL_AND_OUTLINE;
+        var div = document.createElement('div');
+        div.style.position = 'absolute';
+        div.style.opacity = 0;
+        div.style.font = font;
+        document.body.appendChild(div);
+
+        var fontFamily = getCSSValue(div,'font-family');
+        var fontSize = getCSSValue(div,'font-size').replace('px','');
+        var fontStyle = getCSSValue(div,'font-style');
+        var fontWeight = getCSSValue(div,'font-weight');
+
+        document.body.removeChild(div);
+        fontInfo = {
+            fontFamily: fontFamily,
+            fontSize: fontSize,
+            fontStyle: fontStyle,
+            fontWeight: fontWeight
+        };
+        fontInfoCache[font] = fontInfo;
+
+        return fontInfo;
+    }
+
+    var glyphGenerators = {};
+
+    function getGlyphGenerator(fontInfo) {
+        var id = JSON.stringify([
+            fontInfo.fontFamily,
+            fontInfo.fontStyle,
+            fontInfo.fontWeight
+        ]);
+
+        var generator = glyphGenerators[id];
+        if (!defined(generator)) {
+            generator = new TinySDF(glyphFontSize, glyphBuffer, glyphRadius, glyphCutoff, fontInfo.fontFamily, fontInfo.fontWeight, fontInfo.fontStyle);
+            glyphGenerators[id] = generator;
+        }
+        return generator;
+    }
 
-        return writeTextToCanvas(character, writeTextToCanvasParameters);
+    function getSDF(character, fontInfo) {
+        var generator = getGlyphGenerator(fontInfo);
+
+        var canvas = generator.canvas;
+
+        var sdf = generator.draw(character);
+        var ctx = canvas.getContext('2d');
+        var width = ctx.measureText(character).width;
+        var dimensions = {
+            bounds: {
+                minx: 0,
+                maxx: canvas.width,
+                miny: 0,
+                maxy: canvas.height
+            },
+            width: width,
+            height: canvas.height,
+            ascent: 0,
+            descent: 0,
+            fontSize: glyphFontSize
```

Comment:
It seemed that only width and height are needed, unless I'm missing where the other properties are used.

Reply:
The dimension structure here is the same as what was used in the original text implementation.  I'm going to add the ascent and descent calculation in here soon so the kerning will actually work.